src/simplewallet/simplewallet.cpp

```diff
@@ -6955,25 +6972,25 @@ bool simple_wallet::ons_lookup(std::vector<std::string> args)
   int last_index = -1;
   for (auto const &mapping : response)
   {
-    auto& enc_hex = mapping.encrypted_value;
-    if (mapping.entry_index >= args.size() || !oxenmq::is_hex(enc_hex) || enc_hex.size() % 2 != 0 || enc_hex.size() > 2*ons::mapping_value::BUFFER_SIZE)
+    auto enc_hex = mapping["encrypted_value"].get<std::string>();
+    if (mapping["entry_index"].get<uint64_t>() >= args.size() || !oxenmq::is_hex(enc_hex) || enc_hex.size() % 2 != 0 || enc_hex.size() > 2*ons::mapping_value::BUFFER_SIZE)
     {
       fail_msg_writer() << "Received invalid ONS mapping data from oxend";
       return false;
     }
 
     // Print any skipped (i.e. not registered) results:
-    for (size_t i = last_index + 1; i < mapping.entry_index; i++)
+    for (size_t i = last_index + 1; i < mapping["entry_index"]; i++)
       fail_msg_writer() << args[i] << " not found\n";
-    last_index = mapping.entry_index;
+    last_index = mapping["entry_index"];
 
-    const auto& name = args[mapping.entry_index];
+    const auto& name = args[mapping["entry_index"]];
     ons::mapping_value value{};
     value.len = enc_hex.size() / 2;
     value.encrypted = true;
     oxenmq::from_hex(enc_hex.begin(), enc_hex.end(), value.buffer.begin());
 
-    if (!value.decrypt(name, mapping.type))
+    if (!value.decrypt(name, mapping["type"]))
```

Comment:
Unsure if this will decrypt correctly, does nlohmann convert correctly to an enum here?